iceberg/streams/iceberg/schema_name_mapper.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <unordered_set>
+
+#include "iceberg/common/json_parse.h"
+
+namespace iceberg {
+
+class SchemaNameMapper {
+ public:
+  class Node {
+   public:
+    explicit Node(const rapidjson::Value* doc);
+
+    std::optional<int32_t> GetFieldIdByName(const std::string& name) const;
+
+    // TODO: support GetChildNode(const std::string& name)
+
+   private:
+    // Checks that the structure is correct and all names at this level are unique
+    void Validate() const;
+
+    const rapidjson::Value* doc_;
```

Comment:
For now there's no protection that doc_ would be destroyed after the Node.
Place std::weak_ptr<rapidjson::Value> here or make another protection.